subsys/logging/log_backend_rtt.c

```diff
@@ -320,6 +320,8 @@ static void process(const struct log_backend *const backend,
 {
 	uint32_t flags = log_backend_std_get_flags();
 
+	flags |= IS_ENABLED(CONFIG_LOG_BACKEND_UART_SYST_ENABLE) ? LOG_OUTPUT_FLAG_FORMAT_SYST : 0;
```

Comment:
this is not the right place for this. You should move this check and modify the flags in `log_backend_std_get_flags()`

Reply:
@ceolin  std_get_flags() routine is common for all backends and incase 2 backends are enabled at same time, we want to make sure setting of syst format happens outside the std_get_flags() function.